packages/codemirror/src/editor.ts

```diff
@@ -716,6 +716,13 @@ class CodeMirrorEditor implements CodeEditor.IEditor {
        doc.replaceRange('', from, to);
        break;
      case 'set':
+       // Detect the line endings in the text and update
```

Comment:
Should this be set just once when the editor is constructed?

Reply:
The document text is loaded asynchronously

Reply:
Do we not even check at construction whether the string object is empty?

Reply:
here: https://github.com/blink1073/jupyterlab/blob/259ff7aaf25261526d3252a2ab31d5cb2b397deb/packages/codemirror/src/editor.ts#L112

Reply:
True, we need to check in the constructor as well